ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -1320,6 +1320,18 @@ KEXT_STATIC bool CurrentProcessWasSpawnedByRegularUser()
         proc_rele(process);
     }
     
+    if (!nonServiceUser)
+    {
+       // We need to allow the system process amfid to run
+       // It is required for 'open project.app'
```

Comment:
`amfid` is responsible for code signature checks, including on dynamic libraries/frameworks. And this is where the problem was - the dynamic linker `dyld` detected a framework dependency of the app, and so queried `amfid` for the code signing status of the framework. The framework wasn't hydrated, and as `amfid` runs as root, it was blocked from accessing the framework's files instead of hydrating them.

So perhaps we can be more specific in the comment:

```
// When amfid is invoked to check the code signature of a library which has not been hydrated,
// blocking hydration would cause the launch of an application which depends on the library to fail,
// so amfid should always be allowed to hydrate.
```